rfc/rfc003-package-metrics.md

```diff
@@ -12,8 +12,54 @@ http://peter.bourgon.org/go-kit/#package-metrics
 
 ## Scope
 
-To be defined.
+- Package metrics SHALL implement Gauges, Counters, and Histograms.
+
+- Each metric type SHALL allow observations with an unlimited number of key/value field pairs,
+  similar to [package log](https://github.com/peterbourgon/gokit/blob/master/rfc/rfc004-package-log.md).
+
+- Gauge SHALL be an arbitrarily-settable register of type float64.
+
+- Counter SHALL be an increment-only counter of type uint64.
+
+- Histogram SHALL collect observations of type time.Duration.
+
+- These interfaces SHALL be the primary and exclusive API for metrics.
+
+- We SHALL provide a variety of implementations of each interface that act as a
+  bridge to different backends: expvar, Graphite, statsd, Prometheus, etc.
+
+- Each metric backend MAY provide additional value-add behaviors.  For example,
+  a backend for Histogram may bucket observations according to quantile and
+  calculate additional, derived statistics.
+
 
 ## Implementation
 
-To be defined.
\ No newline at end of file
+### Gauge
+
+```go
+type Gauge interface {
+	With(fields ...Fields) Gauge
+	Set(value float64)
+	Inc(delta float64)
+}
+```
+
+### Counter
+
+```go
+type Counter interface {
+	With(fields ...Fields) Counter
+	Inc(delta uint64)
+}
+```
+
+### Histogram
+
+```go
+type Histogram interface {
+	With(fields ...Fields) Histogram
+	Observe(time.Duration)
```

Comment:
Field is just map[string]interface{}, same as package log. Implementation detail.